assertion-analyser.js

```diff
@@ -60,10 +62,10 @@ function objParser(str, init) {
 
 function replacer(str) {
   // replace objects with a symbol ( __#n)
-  var obj;
-  var cnt = 0;
-  var data = [];
-  while(obj = objParser(str)) {
+  let obj;
+  let cnt = 0;
+  let data = [];
+  while(obj == objParser(str)) {
     data[cnt] = obj.obj;
     str = str.substring(0, obj.start) + '__#' + cnt++ + str.substring(obj.end+1)
   }
```

Comment:
```suggestion
  let cnt = 0;
  let data = [];
  let obj = objParser(str);
  while(obj) {
    data[cnt] = obj.obj;
    str = str.substring(0, obj.start) + '__#' + cnt++ + str.substring(obj.end+1);
    objParser(str);
  }
```
I haven't had chance to test it locally, sorry, I will as soon as I can.  For now, here's the suggestion.